packages/spatial/src/physics/classes/Physics.test.ts

```diff
@@ -361,3 +362,714 @@ describe('Physics', () => {
+        setComponent(testEntity, RigidBodyComponent, { type: BodyTypes.Dynamic })
+        Physics.createRigidBody(testEntity, physicsWorld!)
+      })
+
+      afterEach(() => {
+        removeEntity(testEntity)
+        physicsWorld = undefined
+        return destroyEngine()
+      })
+
+      it("should set the body's Translation to the given Position", () => {
+        Physics.setRigidbodyPose(testEntity, position, rotation, linVel, angVel)
+        const body = Physics._Rigidbodies.get(testEntity)!
+        assert.deepEqual(body.translation(), position)
+      })
+
+      /*
+    // @todo How to check rotations?
+    it("should set the body's Rotation to the given value", () => {
+      Physics.setRigidbodyPose(testEntity, position, rotation, linVel, angVel)
+      const body = Physics._Rigidbodies.get(testEntity)!
+      assert.deepEqual(body.rotation(), rotation)
+    })
+    */
+
+      it("should set the body's Linear Velocity to the given value", () => {
+        Physics.setRigidbodyPose(testEntity, position, rotation, linVel, angVel)
+        const body = Physics._Rigidbodies.get(testEntity)!
+        assert.deepEqual(body.linvel(), linVel)
+      })
+
+      it("should set the body's Angular Velocity to the given value", () => {
+        Physics.setRigidbodyPose(testEntity, position, rotation, linVel, angVel)
+        const body = Physics._Rigidbodies.get(testEntity)!
+        assert.deepEqual(body.angvel(), angVel)
+      })
+    })
+
+    describe('enabledCcd', () => {
+      let testEntity = UndefinedEntity
+      let physicsWorld: World | undefined = undefined
+
+      beforeEach(async () => {
+        createEngine()
+        await Physics.load()
+        physicsWorld = Physics.createWorld()
+        getMutableState(PhysicsState).physicsWorld!.set(physicsWorld!)
+        physicsWorld!.timestep = 1 / 60
+
+        // Create the entity
+        testEntity = createEntity()
+        setComponent(testEntity, TransformComponent)
+        setComponent(testEntity, RigidBodyComponent, { type: BodyTypes.Dynamic })
+        Physics.createRigidBody(testEntity, physicsWorld!)
+      })
+
+      afterEach(() => {
+        removeEntity(testEntity)
+        physicsWorld = undefined
+        return destroyEngine()
+      })
+
+      it('should enable Continuous Collision Detection on the entity', () => {
+        const body = Physics._Rigidbodies.get(testEntity)!
+        assert.equal(body.isCcdEnabled(), false)
+        Physics.enabledCcd(testEntity, true)
+        assert.equal(body.isCcdEnabled(), true)
+      })
+
+      it('should disable CCD on the entity when passing `false` to the `enabled` property', () => {
+        const body = Physics._Rigidbodies.get(testEntity)!
+        Physics.enabledCcd(testEntity, true)
+        assert.equal(body.isCcdEnabled(), true)
+        Physics.enabledCcd(testEntity, false)
+        assert.equal(body.isCcdEnabled(), false)
+      })
+    })
+
+    /**
+  // @todo How to check for the impulse of an entity?
```

Comment:
impulse will have a resultant velocity the next frame, we can calculate what the velocity is, or just assert it's beforehand and assert it's not zero or close to an expected value afterwards

Reply:
```ts
it("should apply the impulse to the RigidBody of the entity", () => {
  const testImpulse = new Vector3(1,2,3)
  const body = Physics._Rigidbodies.get(testEntity)
  assert.ok(body)
  
  console.log(JSON.stringify(body.linvel()))
  console.log(JSON.stringify(body.angvel()))
  
  Physics.applyImpulse(testEntity, testImpulse)
  
  console.log(JSON.stringify(body.linvel()))
  console.log(JSON.stringify(body.angvel()))
})
```
```ts
// Output
{"x":0,"y":0,"z":0}
{"x":0,"y":0,"z":0}
{"x":0,"y":0,"z":0}
{"x":0,"y":0,"z":0}
```
Am I missing something to "sync" the physics so that they are applied right away?

Reply:
you need to run the physics simulation, looks at PhysicsSystem.ts

Reply:
Fixed somewhere in the commit list of the PR
_Forgot to track the commit where I applied it, and now I cant find the sha_